packages/playground/src/components/account/account-exchange/account-exchange.tsx

```diff
@@ -47,18 +52,40 @@ export class AccountExchange {
 
   async onDepositClicked(e) {
     const amount = e.target.value;
-    await this.deposit(amount, this.user.multisigAddress);
+    try {
+      await this.deposit(ethers.utils.parseEther(amount));
+    } catch (e) {
+      if (e.toString().includes("Cannot deposit while another deposit")) {
+        window.alert(HUB_IS_DEPOSITING_ALERT);
```

Comment:
@snario We have a `<widget-dialog>` component to take care of this sort of messages. We should render that instead of using `window.alert()`.

You can define a `@State() dialog: JSX.Element` member in the component class and have it set to something like:

```jsx
const content = <label>{HUB_IS_DEPOSITING_ALERT}</label>;

this.dialog = <widget-dialog
  visible={true}
  dialogTitle="Sorry, can't deposit right now :("
  content={content}
  primaryButtonText="OK"
  onPrimaryButtonClicked={() => this.dialog = undefined}
/>;
```

Then, you should add `{this.dialog}` to the `render()` JSX.

Reply:
This was actually meant to be a temporary stub and my plan was to listen for the deposit events from the playground-server and disable the entire /exchange UI and replace with an overlay that would say "Please wait for the hub to collateralize your channel." and some secondary text "Curious why you can't deposit? See <this issue> on the Counterfactual GitHub if you're curious to understand how asynchronous deposits will be added in the future"